File: SwingApps/src/cc/lib/swing/AWTLabel.java
package cc.lib.swing;

import javax.swing.*;
import java.awt.*;

import cc.lib.game.Utils;

public class AWTLabel extends JLabel {

    /**
     *
     * @param text
     * @param justify 0 == left, 1 == center, 2 == right
     * @param size
     * @param bold
     */
    public AWTLabel(String text, int justify, float size, boolean bold) {
        super(text);
        switch (justify) {
            case 0:
                setHorizontalAlignment(SwingConstants.LEFT);
                break;
            case 1:
                setHorizontalAlignment(SwingConstants.CENTER);
                break;
            case 2:
                setHorizontalAlignment(SwingConstants.RIGHT);
                break;
            default:
                Utils.unhandledCase(justify);
        }
        Font font = getFont().deriveFont(bold ? Font.BOLD : Font.PLAIN, size);
        setFont(font);
    }

}

File: SwingApps/src/cc/lib/swing/AWTLabel.kt
package cc.lib.swing

import cc.lib.game.Utils
import java.awt.Font
import javax.swing.JLabel

class AWTLabel(text: String?, justify: Int, size: Float, bold: Boolean) : JLabel(text) {
	/**
	 *
	 * @param text
	 * @param justify 0 == left, 1 == center, 2 == right
	 * @param size
	 * @param bold
	 */
	init {
		when (justify) {
			0 -> horizontalAlignment = LEFT
			1 -> horizontalAlignment = CENTER
			2 -> horizontalAlignment = RIGHT
			else -> Utils.unhandledCase(justify)
		}
		val font = font.deriveFont(if (bold) Font.BOLD else Font.PLAIN, size)
		setFont(font)
	}
}